components/Dialogs/src/MDCDialogPresentationController.m

```diff
@@ -301,7 +301,17 @@ - (void)preferredContentSizeDidChangeForChildContentContainer:(id<UIContentConta
 
 - (void)dismiss:(UIGestureRecognizer *)gesture {
   if (gesture.state == UIGestureRecognizerStateRecognized) {
-    [self.presentingViewController dismissViewControllerAnimated:YES completion:NULL];
+    __weak __typeof(self) weakSelf = self;
+    [self.presentingViewController
+        dismissViewControllerAnimated:YES
+                           completion:^{
+                             if ([weakSelf.dialogPresentationControllerDelegate
```

Comment:
Take a strong reference to the weakSelf at the beginning of this block so that the two invocations are made on the same object instance.

Reply:
@jverkoey, by "Take a strong reference to the weakSelf" do you just mean "don't use weakSelf"? 
I don't usually run the static analyzer actually, but I just did, and it didn't take any issues with this use of weakSelf.
I'll get rid of the weakSelf!

Reply:
Also, I wouldn't expect the block to require a weak reference here. Since animation blocks in UIKit are not copied with retained references, it should be safe to use `self` directly.

Reply:
I don't always do strongSelf because sometimes it's just okay to send a message to nil. But yeah there probably isn't any risk of a retain cycle here, I just once had a boss who was very insistent on any self in a block being weakified and now it's a habit 🤷‍♂️

Reply:
Also, to be consistent with our code base, please don't use `typeof`. Instead, make the pointer explicitly the class type.  If you search `__weak` in our codebase you'll see that the vast majority are explicitly typed.